vispy/visuals/volume.py

```diff
@@ -483,7 +518,40 @@ def __init__(self, vol, clim=None, method='mip', threshold=None,
         self.relative_step_size = relative_step_size
         self.threshold = threshold if (threshold is not None) else vol.mean()
         self.freeze()
-    
+
+    def _create_texture(self, texture_format, emulate_texture, data):
+        tex_cls = TextureEmulated3D if emulate_texture else Texture3D
+
+        tex_kwargs = {}
+        if isinstance(texture_format, str) and texture_format == 'auto':
+            texture_format = data.dtype.type
+        if texture_format and not isinstance(texture_format, str):
+            if texture_format not in self._texture_dtype_format:
+                raise ValueError("Can't determine internal texture format for '{}'".format(texture_format))
+            texture_format = self._texture_dtype_format[texture_format]
+        if isinstance(texture_format, str):
+            tex_kwargs['internalformat'] = texture_format
+            tex_kwargs['format'] = 'luminance'
+
+        # Use 3D array shape of (10, 10, 10) as placeholder. When data is
+        # set later this will be resized.
+        # clamp_to_edge means any texture coordinates outside of 0-1 should be
+        # clamped to 0 and 1.
+        return tex_cls((10, 10, 10), interpolation=self._interpolation,
+                       wrapping='clamp_to_edge', **tex_kwargs)
+
+    def _cpu_scale_data(self, vol):
+        if self._clim[1] == self._clim[0]:
+            if self._clim[0] != 0.:
+                vol *= 1.0 / self._clim[0]
+        elif self._clim[0] > self._clim[1]:
+            vol *= -1
+            vol += self._clim[1]
+            vol /= self._clim[1] - self._clim[0]
```

Comment:
`self._clim[1] - self._clim[0]` is negative and will reverse sign of vol, is that on purpose ?

Reply:
Yes I think so. It has been a while since I've looked at this code, but @tlambert03 and I had a long discussion about this in one of his PRs. There was something funky going on with the operations and what the data ended up as in the shader that we couldn't track down at the time. This was just the way we/he ended up implementing it and it worked so we kept it. The multiply by `-1` at the top and then divide by this negative result produces the expected result. I'll try re-evaluating after I've dealt with the ImageVisual PR and added real tests to this PR.

Reply:
yeah, I agree it looks weird.  I can't remember the details at this point... but I definitely spent a while trying to make it look like it "should" without success.  no reason not to try again though!